### PR TITLE
added numpy.choose frontend function

### LINKED ISSUE #23331: choose

### PR DESCRIPTION
**Pull request by OriginCoding**
feat: added the numpy.choose instance method

## Related Issue

Close #23331 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by OriginCoding** [comment]
numpy.choose is absent in backend but i have implemented in frontend

**Comment by OriginCoding** [comment]
Its almost a week can it get reviewed early

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @OriginCoding 

Apologies for the delay in reviewing your PR. Have been swamped with work recently and so haven't had the time to review any PRs 😅.

Regarding the PR itself, I am noticing that the tests are failing due to a `key error` [here](https://github.com/unifyai/ivy/actions/runs/6246782847/job/16958054722?pr=23335#step:3:166). 

Could you kindly look into this? Also, while you're at it, could you kindly ensure the following checks also pass: 
- [linting tests](https://github.com/unifyai/ivy/actions/runs/6246782889/job/16958055101?pr=23335#step:5:155)
- [PR title convention](https://github.com/unifyai/ivy/actions/runs/6247054211/job/16958906837?pr=23335#step:2:43) 

Thanks! 🙂

**Comment by OriginCoding** [comment]
i tried but not able to solve this error also edited the PR description

File: ivy/functional/frontends/numpy/creation_routines/from_existing_data.py [hunk 1]
```diff
@@ -34,3 +34,8 @@ def copy(a, order="K", subok=False):
 @handle_numpy_dtype
 def frombuffer(buffer, dtype=float, count=-1, offset=0, *, like=None):
     return ivy.frombuffer(buffer)
+
+
+@to_ivy_arrays_and_back
+def choose(arr, choices, mode=None):
+    return ivy.choose(arr=arr, choices=choices, mode=mode)
```
Comment (YushaArif99) [inline ivy/functional/frontends/numpy/creation_routines/from_existing_data.py:41]: I don't think you should be passing all arguments as `kwargs` when calling `ivy.choose`. Perhaps this is causing the error [here](https://github.com/unifyai/ivy/actions/runs/6246782847/job/16958054722?pr=23335#step:3:231)

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @OriginCoding 

The CI tests are still failing with a different error this time 😅. Kindly view the above comment for more detail on this. 
Also, the PR compliance checks are *still* failing [here](https://github.com/unifyai/ivy/actions/runs/6255593787/job/16985174936?pr=23335#step:2:43). Could you kindly resolve these as well?
Edit: The linting tests are also failing. Could you kindly run `pre-commit` on the modified files and push the changes? This should resolve the linting errors.
Thanks! 🙂

**Pull request closed without merging**